trace/opencensus/trace/propagation/text_format.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module provides the basic utilities for extracting the trace
+information from a carrier which is a dict to form a SpanContext. And
+generating a dict using the provided SpanContext.
+"""
+
+from opencensus.trace.span_context import SpanContext
+
+_OPENCENSUS_TRACE_PREFIX = 'opencensus-trace'
+_TRACE_ID_KEY = '{}-traceid'.format(_OPENCENSUS_TRACE_PREFIX)
+_SPAN_ID_KEY = '{}-spanid'.format(_OPENCENSUS_TRACE_PREFIX)
+_ENABLED_TRACE_KEY = '{}-enabled'.format(_OPENCENSUS_TRACE_PREFIX)
+
+
+def from_carrier(carrier):
+    """Generate a SpanContext object using the information in the carrier.
+
+    :type carrier: dict
+    :param carrier: The carrier which has the trace_id, span_id, options
+                    information for creating a SpanContext.
+
+    :rtype: :class:`~opencensus.trace.span_context.SpanContext`
+    :returns: SpanContext generated from the carrier.
+    """
+    trace_id = None
+    span_id = None
+    enabled = 1
+
+    for key in carrier:
+        key = key.lower()
+        if key == _TRACE_ID_KEY:
+            trace_id = carrier[key]
+        if key == _SPAN_ID_KEY:
+            span_id = carrier[key]
+        if key == _ENABLED_TRACE_KEY:
+            enabled = bool(carrier[key])
+
+    return SpanContext(
```

Comment:
The default value of trace_id is None, so if it doesn't exist in the carrier, it will just be None. And the constructor of SpanContext will generate a new trace_id for it.